src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -474,6 +474,66 @@ protected override TextReader CreateTextReaderImplementation()
             return reader;
         }
 
+        /// <summary>
+        /// Asynchronously creates <see cref="Stream"/> for reading an inline stream property.
+        /// </summary>
+        /// <returns>
+        /// A task that represents the asynchronous operation.
+        /// The value of the TResult parameter contains a <see cref="Stream"/> for reading the stream property.
+        /// </returns>
+        protected override async Task<Stream> CreateReadStreamImplementationAsync()
+        {
+            Stream stream;
+            IJsonStreamReaderAsync streamReader = this.jsonLightInputContext.JsonReader as IJsonStreamReaderAsync;
+            if (streamReader != null)
+            {
+                stream = await streamReader.CreateReadStreamAsync()
+                    .ConfigureAwait(false);
+            }
+            else
+            {
+                // JsonReader doesn't support streaming; read as a string and convert
+                // Skip over property or start array
+                await this.jsonLightInputContext.JsonReader.ReadAsync()
+                    .ConfigureAwait(false);
+                string valueAsString = await this.jsonLightInputContext.JsonReader.ReadStringValueAsync()
+                    .ConfigureAwait(false);
+                stream = new MemoryStream(Convert.FromBase64String(valueAsString.Replace('_', '/').Replace('-', '+')));

```

Comment:
Are these `Replace` calls some escaping logic? I'm not familiar with base64 being used that way. Do we ever go the other direction, replacing '/' with '_' etc. ?